identity-appsupport/src/commonMain/kotlin/com/android/identity/appsupport/ui/certificateviewer/CertificateViewer.kt

```diff
@@ -0,0 +1,406 @@
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun PreviewCertificate() {
+    val key = Crypto.createEcPrivateKey(EcCurve.P256)
+    val now = Instant.fromEpochSeconds(Clock.System.now().epochSeconds)
+    val x509 = X509Cert.Builder(
+        publicKey = key.publicKey,
+        signingKey = key,
+        signatureAlgorithm = key.curve.defaultSigningAlgorithm,
+        serialNumber = ASN1Integer(1),
+        subject = X500Name.fromName("CN=Foobar1"),
+        issuer = X500Name.fromName("CN=Foobar2"),
+        validFrom = now - 1.hours,
+        validUntil = now + 1.hours
+    )
+        .includeSubjectKeyIdentifier()
+        .includeAuthorityKeyIdentifierAsSubjectKeyIdentifier()
+        .build()
+
+    CertificateView(from(x509), null, null)
+}
+
+/**
+ * Verify supported use cases, throw with detailed message on failure. Parameters are passed in
+ * partially reversed mode to make adding new certificate objets easier (to the list end).
+ */
+private fun validateParameters(
+    infos: List<String>?,
+    warnings: List<String>?,
+    x509CertChain: X509CertChain?,
+    x509Cert: X509Cert?,
+    // Add more types as needed here.
+) {
+    val certificatesPassed = listOfNotNull(
+        x509CertChain,
+        x509Cert,
+        // Add more types as needed here.
+    )
+    if (certificatesPassed.isEmpty()) {
+        throw IllegalArgumentException("No certificates provided.")
+    }
+    if (certificatesPassed.size > 1) {
+        throw IllegalArgumentException("Only one certificate object should be provided.")
+    }
+
+    // Test chain size to match warnings and infos sizes.
+    if (x509CertChain != null) {
+        val listSize = x509CertChain.certificates.size
+        if (infos != null && infos.size != listSize || warnings != null && warnings.size != listSize) {
+            val message = buildString {
+                append("The list size of ")
+                if (infos != null && infos.size != listSize) {
+                    append("infos")
+                    if (warnings != null && warnings.size != listSize) {
+                        append(" and warnings")
+                    }
+                } else if (warnings != null && warnings.size != listSize) {
+                    append("warnings")
+                }
+                append(" parameters provided, must be equal to the certificates chain size.")
+            }
+            throw IllegalArgumentException(message)
+        }
+    }
+}
+
+private fun resFromKey(it: String): StringResource {
+    val nMap = mapOf(
+        OID.COMMON_NAME.oid to Res.string.k_common_name,
+        OID.COUNTRY_NAME.oid to Res.string.k_country_name,
+        OID.LOCALITY_NAME.oid to Res.string.k_locality_name,
+        OID.STATE_OR_PROVINCE_NAME.oid to Res.string.k_state_name,
+        OID.ORGANIZATION_NAME.oid to Res.string.k_org_name,
+        OID.ORGANIZATIONAL_UNIT_NAME.oid to Res.string.k_org_unit_name,
```

Comment:
Maybe add `// TODO: Add support for other OIDs from RFC 5280 Annex A, as needed.`